transformer.md

```diff
@@ -255,3 +255,62 @@ The end marker is the following, where _`<N>`_ is the number of the step:
 ```markdown
 <!-- Killercoda step<N>.md END -->
 ```
+
+## Generate a tutorial
+
+You can generate a new tutorial from existing documentation using the transformer tool.
+After generation, a [GitHub Actions workflow](./.github/workflows/regenerate-tutorials) updates the tutorial when the documentation source changes.
+
+### Before you begin
+
+1. Clone the repository with the source documentation.
+1. Clone the Killercoda repository.
+1. [Download and install Go](https://go.dev/doc/install) to run the transformer tool.
+
+To generate a tutorial:
+
+1. In the source repository, check out a new branch.
+
+   Give the branch a name that reflects the planned changes.
+   For example, `2024-06-killercoda-tutorial-for-loki-quickstart`.
+
+1. In the Killercoda repository, check out a new branch.
+
+   For convenience, name the branch the same as you did in the source repository.
+
+1. In the source repository, run `make docs` from the `docs/` directory so that you can verify your changes don't break the rendered documentation.
+
+1. In the source file, add the Killercoda front matter.
+
+   For more information and an example front matter, refer to [Metadata](#Metadata)
+
+1. Add directives for each step in the tutorial.
+
+   Each step starts at the [step](#step) directive start marker and ends at the [step](#step) directive end marker.
+   Include at least one step.
+   The first step use the start marker `<!-- Killercoda step1.md START -->` and the end marker `<!-- Killercoda step1.md END -->`
+
+1. Generate the tutorial.
+
+   1. In the Killercoda repository, change to the `tools/transformer` directory.
+   1. Run `go run ./ <PATH TO SOURCE FILE> <PATH TO OUTPUT DIRECTORY>`
+
+      - _`<PATH TO SOURCE FILE>`_ is the path to the documentation file.
+
+        For example, `~/ext/grafana/loki/docs/sources/get-started/quick-start.md`.
+
+      - _`<PATH TO OUTPUT DIRECTORY>`_ is the path to the output directory in the Killercoda repository.
+
+        For example, `~/ext/grafana/killercoda/loki/loki-quickstart`.
+
+      The tool creates the `index.json` and step Markdown files.
+
+1. Add the tutorial to the Killercoda structure file.
+
+   The structure file is a `structure.json` file in each directory that contains tutorials.
+
+   The Loki structure file is [`loki/structure.json`](./loki/structure.json).
+
+1. In each repository, commit your changes, push your branch, and open a pull request.
+
+1. A Killercoda maintainer reviews the PR to ensure that the generate tutorial works as expected.
```

Comment:
Noice shall we also ask them to add it to the github workflow or we as maintainers should do this?

Reply:
I think for now, we may need to manage that because the workflow isn't super friendly to new contributors. 

I can write a separate doc for the task of updating the regeneration workflow and perhaps that will make it easier to contribute to or will force me to refactor the workflow to be easier for contributors to modify.